pkg/lsp/service.go

```diff
@@ -38,8 +38,20 @@ const (
 	TDSKIncremental                      = 2
 )
 
+type TextDocumentSyncOptions struct {
+	OpenClose         bool                 `json:"openClose,omitempty"`
+	Change            TextDocumentSyncKind `json:"change"`
+	WillSave          bool                 `json:"willSave,omitempty"`
+	WillSaveWaitUntil bool                 `json:"willSaveWaitUntil,omitempty"`
+	Save              *SaveOptions         `json:"save,omitempty"`
+}
+
+type SaveOptions struct {
+	IncludeText bool `json:"includeText"`
+}
+
 type ServerCapabilities struct {
-	TextDocumentSync                 int                              `json:"textDocumentSync,omitempty"`
+	TextDocumentSync                 TextDocumentSyncOptions          `json:"textDocumentSync,omitempty"`
```

Comment:
Update: After my latest commit, we can update anything in any order, as long as we don't change anything to send  `TextDocumentSyncOptions` (need to keep it just sending the `TextDocumentSyncKind`). Once all are upgraded, then we can change them to send the `TextDocumentSyncOptions` in any order.